accounts/abi/bind/base.go

```diff
@@ -49,10 +49,14 @@ type TransactOpts struct {
 	Nonce  *big.Int       // Nonce to use for the transaction execution (nil = use pending state)
 	Signer SignerFn       // Method to use for signing the transaction (mandatory)
 
-	Value    *big.Int // Funds to transfer along the transaction (nil = 0 = no funds)
-	GasPrice *big.Int // Gas price to use for the transaction execution (nil = gas price oracle)
+	Value    *big.Int // Funds to transfer along along the transaction (nil = 0 = no funds)
 	GasLimit uint64   // Gas limit to set for the transaction execution (0 = estimate)
 
+	// If GasPrice, GasPremium, and FeeCap are all nil then we defer to the gas price oracle
+	GasPrice            *big.Int // Gas price to use for the transaction execution
```

Comment:
"legacy" can be added to the comment: "_Gas price to use for the legacy transaction execution_"